src/nnet2/nnet-component.cc

```diff
@@ -4110,6 +4124,8 @@ void Convolutional1dComponent::Read(std::istream &is, bool binary) {
   ReadBasicType(is, binary, &patch_step_);
   ExpectOneOrTwoTokens(is, binary, ostr_beg.str(), "<PatchStride>");
   ReadBasicType(is, binary, &patch_stride_);
+  ExpectOneOrTwoTokens(is, binary, ostr_beg.str(), "<RearrangeInput>");
```

Comment:
Vijay, it's not clear to me that this explanation can be done in the
variable very easily... even if you talk about zyx and yzx, it would take
some explanation to express what this really means (it's certainly not
clear to me).  So for me it would be find to leave the variable name as it
is, and put this explanation somewhere in the documentation.
Unless it's the case that we will most likely later want to support other
kinds of rearrangement and having a boolean variable is too limiting.
Dan


On Mon, Oct 26, 2015 at 2:00 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In src/nnet2/nnet-component.cc
> <https://github.com/kaldi-asr/kaldi/pull/296#discussion_r43029383>:
>
> > @@ -4110,6 +4124,8 @@ void Convolutional1dComponent::Read(std::istream &is, bool binary) {
> >    ReadBasicType(is, binary, &patch_step_);
> >    ExpectOneOrTwoTokens(is, binary, ostr_beg.str(), "<PatchStride>");
> >    ReadBasicType(is, binary, &patch_stride_);
> > +  ExpectOneOrTwoTokens(is, binary, ostr_beg.str(), "<RearrangeInput>");
>
> I would recommend using variable names which do not require the user to
> read the code to understand their usage. I would recommend using a variable
> which specifies in detail what is the default vectorization order and when
> users would have to rearrange_input.
>
> I tried to solve this issue by using input_vectorization variable in
> ConvolutionComponent being written for nnet3. (see #283
> <https://github.com/kaldi-asr/kaldi/pull/283>, @line
> <https://github.com/vijayaditya/kaldi/blob/05b00b8834d3c45baed4461dc923109da994c3cb/src/nnet3/nnet-simple-component.h#L922>
> ) The documentation for this component is not complete, but I basically
> tried to explain that zyx is the default vectorization order and users can
> specify that they are using yzx input. Based on this info the component
> decides if it has to rearrange the input or not.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/296/files#r43029383>.
>